components/Stands/StandCard.tsx

```diff
@@ -0,0 +1,224 @@
+import React, { useEffect, useState } from 'react';
+import styled from 'styled-components';
+import { NudgeDiv } from '../Styled';
+import { useRouter } from 'next/router';
+import dayjs, { Dayjs } from 'dayjs';
+
+import customParseFormat from 'dayjs/plugin/customParseFormat';
+import { timeIsBetween, toDayjs } from '../../utils/time';
+import { createFragmentContainer, graphql } from 'react-relay';
+import { StandCard_stand } from '../../__generated__/StandCard_stand.graphql';
+import HSPEvents from './HSPEvents';
+import CompanyCardContent from './CompanyCardInfo';
+import LiveIndicator from './LiveIndicator';
+
+dayjs.extend(customParseFormat);
+
+export type ArrayElement<
+  ArrayType extends readonly unknown[]
+> = ArrayType[number];
+type Package = 'standard' | 'sp' | 'hsp';
+export type standEvents = NonNullable<StandCard_stand['events']>;
+export type standEvent = ArrayElement<standEvents>;
+interface StandCardProps {
+  type: Package;
+  time: Dayjs;
+  stand: StandCard_stand;
+}
+
+export const getCurrentEvent = (
+  events: standEvents | null,
+  time: Dayjs
+): standEvent | null => {
+  const currentEvent = events?.find(
+    (event) =>
+      event &&
+      timeIsBetween({
+        time: time,
+        start: toDayjs(event.date, event.timeStart),
+        end: toDayjs(event.date, event.timeEnd),
+      })
+  );
+  return currentEvent ?? null;
+};
+
+interface EventInfo {
+  timeRange: string;
+  eventTitle: string;
+  eventDescription: string;
+}
+
+const StandCard = ({ stand, time, type }: StandCardProps): JSX.Element => {
+  const [currentEvent, setCurrentEvent] = useState<standEvent | null>();
+  const router = useRouter();
+
+  useEffect(() => {
+    const newCurrentEvent = getCurrentEvent(stand.events ?? [], time);
+    setCurrentEvent(newCurrentEvent);
+  }, [time, stand]);
```

Comment:
Why do you have state and an effect hook here? currentEvent is only dependant on stand and time, which are both props, so it will update automatically without the need of a hook.

Reply:
Hmm.. I guess I could add the `getCurrentEvent` as the default state

Reply:
Or maybe it would suffice with
```suggestion
  const currentEvent = getCurrentEvent(stand.events ?? [], time);
```

Reply:
I'm pretty sure this should do it yes.